### PR TITLE
Model docs & fixes

### PR DESCRIPTION
**Pull request by Shqrdx**
Changes:
-Docs for all members in Model
-Sealed modificator added to Model(as in XNA).
-Added new method SetCustomEffect which can be used for quickly apply custom effect to all meshes.
-Restrict constructors(in original XNA it is impossible to use them directly).
-Added missed XNA members CopyBoneTransformsTo and CopyBoneTransformsFrom.

### CODE REVIEW

**Comment by Shqrdx** [comment]
@tomspilman Your throughts ?

**Comment by Shqrdx** [comment]
@tomspilman Please react.

File: MonoGame.Framework/Graphics/Model.cs [hunk 1]
```diff
@@ -1,44 +1,80 @@
...
 using System;
-using System.Diagnostics;
 using System.Collections.Generic;
-using System.Collections.ObjectModel;
 
 namespace Microsoft.Xna.Framework.Graphics
 {
-	public class Model
+    /// <summary>
+    /// Represents a 3D-Model.
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Model.cs]: This probably needs a little more detail.

> A basic 3D model with per mesh parent bones.

@KonajuGames ?

File: MonoGame.Framework/Graphics/Model.cs [hunk 2]
```diff
@@ -1,44 +1,80 @@
...
+        /// foreach (var mesh in model.Meshes)
+        /// {
+        ///    foreach (var part in mesh.MeshParts)
+        ///    {
+        ///       part.Effect = effect;
+        ///    }
+        /// }
+        /// </code>
+        /// </remarks>
+        public void SetCustomEffect(Effect effect)
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Model.cs]: This function seems completely unnecessary to me.  Especially we should not copy the code from a function into the function's own documentation.

@KonajuGames ?

File: MonoGame.Framework/Graphics/Model.cs [hunk 3]
```diff
@@ -75,8 +111,14 @@ private void BuildHierarchy(ModelBone node, Matrix parentTransform, int level)
 			//
 			//Debug.WriteLine("{0}:{1}", s, node.Name);
 		}
-		
-		public void Draw(Matrix world, Matrix view, Matrix projection) 
+
+        /// <summary>
+        /// Draws a model after applying the matrix transformations.
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Model.cs]: > Draw the model meshes.

File: MonoGame.Framework/Graphics/Model.cs [hunk 4]
```diff
@@ -75,8 +111,14 @@ private void BuildHierarchy(ModelBone node, Matrix parentTransform, int level)
...
 		}
-		
-		public void Draw(Matrix world, Matrix view, Matrix projection) 
+
+        /// <summary>
+        /// Draws a model after applying the matrix transformations.
+        /// </summary>
+        /// <param name="world">A matrix of world transformation.</param>
+        /// <param name="view">A matrix of view transformation.</param>
+        /// <param name="projection">A matrix of projection transformation.</param>
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Model.cs]: > A matrix of projection transformation

That doesn't sound right in English.  Say `The projection transform.`

File: MonoGame.Framework/Graphics/Model.cs [hunk 5]
```diff
@@ -106,8 +148,12 @@ public void Draw(Matrix world, Matrix view, Matrix projection)
                 mesh.Draw();
             }
 		}
-		
-		public void CopyAbsoluteBoneTransformsTo(Matrix[] destinationBoneTransforms)
+
+        /// <summary>
+        /// Copies bone transforms relative to all parent bones of the each bone from this model to a given array.
+        /// </summary>
+        /// <param name="destinationBoneTransforms">The array to receive transform of bones.</param>
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Model.cs]: `The array to receive the transformed bones.`

File: MonoGame.Framework/Graphics/Model.cs [hunk 6]
```diff
@@ -128,70 +174,41 @@ public void CopyAbsoluteBoneTransformsTo(Matrix[] destinationBoneTransforms)
...
+            for (int i = 0; i < count; i++)
+            {
+                Bones[i].Transform = sourceBoneTransforms[i];
+            }
+        }
+
+        /// <summary>
+        /// Copies bone transforms relative to <see cref="Model.Root"/> bone from this model to a given array.
+        /// </summary>
+        /// <param name="destinationBoneTransforms">The array to receive transform of bones.</param>
```
Comment (tomspilman) [inline MonoGame.Framework/Graphics/Model.cs]: >  The array receiving the transformed bones.

**Comment by Shqrdx** [comment]
@tomspilman Whats the problem ? Windows 10 ?

**Comment by Shqrdx** [comment]
I'l close it and remake.

**Comment by tomspilman** [comment]
@Shqrdx - Yeah... we added Win10 support and the build process is complaining it can't find the Win10 build step in the NAnt script.  Just merge/rebase the latest and it will be fixed.

**Comment by Shqrdx** [comment]
@tomspilman I remaded it. Please review.

**Pull request closed without merging**